### ISSUE TITLE
Teleporting into Betweenland doesnt work correct/bugs away/out - Sevtech Ages of the Sky

### ISSUE THREAD

**Issue by AtPMonster** [DarkPacks/SevTech-Ages#4301]
## Bug Description

 Hello its me again!

So I travvelled to the betweenlands/twilight for the first time and see what happens.. Im bugging out in nothingness. Then I reloaded and I was suddenly down in a really deep cave (not in the twilight). Had to get back to the surface and somehow survived - without seeing the portal to get back into my own world :-/ Lucky I had the amulet with me to create a new one with a Sapling. Still hard to get that.. Anyways, I got back into the overworld easy, but at a really random spot. So I had to activate Creative to check the area out and lucky point, it worked and I got back to my base.

You can check my video for that.

EDIT: I crafted both portals at a new spot, and that seemed to fix it. You cant see the problem with the Twilight in these videos yet.

### Did this behavior use to work in the previous version?

 No, it didnt in Sevtech Ages (Non skyblock)

## Steps to Reproduce (for bugs)
 Twitch Video:  https://www.twitch.tv/atpgamestreams/about or https://www.twitch.tv/videos/751819092 - starting from 18:05 min and keeps happening around the whole video again.

1. Get into the Portal
2. Wait
3. Spawn somewhere in the nothingness
4. Reload
5. Spawn somewhere in a cave deep down.

## Logs

* Client/Server Log:
* Crash Log:

## Client Information
* Modpack Version: 3.2.0 Alpha 2
* Java Version:
* Launcher Used:Twitch
* Memory Allocated: 11GB
* Server/LAN/Single Player: SP
* Optifine Installed: Yes
* Shaders Enabled: No.

## World Information
* Modpack Version world created in:SevTech: Ages of the Sky 3.2.0 Alpha 2
* Additional Content Installed: Optifine

**Comment by Darkosto** [comment]
My first concern is that you saved and quit the game during loading. Yes, it took 40 seconds before you clicked the button (for reference my PC takes 25-28 seconds but I am also not streaming at the same time so my resources are not being split between the game and broadcasting). I suspect that if you did not quit the game you would not have ended up in the ground. 

In your report, you say you spawned into nothingness. This is how the dimension normally loads. I can confirm this because I've been working with this mod for over 3 years now. That's just what it does.

I would caution you against quitting out of the game during a load unless it's taking a severe amount of time. You should not see anything more than 1-2 minutes. If it's taking that long then I would say something might be wrong. 

I've also noticed you dedicate 11gb of RAM to the game? Is there any specific reason of doing this? If you have been playing the game long enough, it's very possible that the garbage collection would increase the load times. I'd urge you not to dedicate more than 8gb of RAM total to Minecraft UNLESS you absolutely need to due to a resource pack or shader pack. Many times, they do not need much more than 8gb.

You may also benefit from using java arguments in the Twitch launcher to help the game perform slightly better in terms of Garbage Collection: 

`
-Xmx8g -Xms8g -XX:+UseG1GC -Dsun.rmi.dgc.server.gcInterval=2147483646 -XX:+UnlockExperimentalVMOptions -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
`

Give that a try and see if you notice a difference.

**Comment by AtPMonster** [comment]
> My first concern is that you saved and quit the game during loading. Yes, it took 40 seconds before you clicked the button (for reference my PC takes 25-28 seconds but I am also not streaming at the same time so my resources are not being split between the game and broadcasting). I suspect that if you did not quit the game you would not have ended up in the ground.
> 
> In your report, you say you spawned into nothingness. This is how the dimension normally loads. I can confirm this because I've been working with this mod for over 3 years now. That's just what it does.
> 
> I would caution you against quitting out of the game during a load unless it's taking a severe amount of time. You should not see anything more than 1-2 minutes. If it's taking that long then I would say something might be wrong.
> 
> I've also noticed you dedicate 11gb of RAM to the game? Is there any specific reason of doing this? If you have been playing the game long enough, it's very possible that the garbage collection would increase the load times. I'd urge you not to dedicate more than 8gb of RAM total to Minecraft UNLESS you absolutely need to due to a resource pack or shader pack. Many times, they do not need much more than 8gb.
> 
> You may also benefit from using java arguments in the Twitch launcher to help the game perform slightly better in terms of Garbage Collection:
> 
> `-Xmx8g -Xms8g -XX:+UseG1GC -Dsun.rmi.dgc.server.gcInterval=2147483646 -XX:+UnlockExperimentalVMOptions -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M`
> 
> Give that a try and see if you notice a difference.

Using the 11 GB because the game sometimes freezes for a few moments and when I check it, it almost used 99% of allocated ram, thats why I do it. If these are the same augments like for normal Sevtech, then it will make it worse^^ Already tried these

**Comment by Darkosto** [comment]
I'm not sure if what I said previously wasn't communicated well but let me try again:

Are you familiar with Garbage Collection with Java? The freezes you are mentioning are most likely the garbage collection process which is a normal process. The more RAM you dedicate the worse it gets. There will be more RAM that has to be dumped and your freezes will continuously get worse. You think you are fixing the problem by dedicating more but, in reality, it's not helping you. It's like having to clear the cache on a phone or tablet, the larger the cache the more time it will take.  

Seeing the 99% of allocated RAM does not necessarily indicate that you need to increase the RAM. If you watch the RAM using the F3 menu, it will climb up then garbage collection happens. It will fall back down and continue to do this over the course of your playthrough. If you dedicate it, Minecraft will use it. This pack can be run with 4GB and be just fine. Again, I would suggest 8GB. The memory is reserved when you dedicate it. Meaning, it will show as being a higher percentage but that's not indicative of you having to give it more.

To further clarify, those arguments are written by someone who not only works on the modding API (Forge) for minecraft but also works at the company that makes Java. I would urge you to read how the arguments work and learn a bit more about them because I think you are misunderstanding what they are doing. Of course, not every solution will work for everyone. Having said that, these arguments are meant to fix the EXACT problem you are experiencing with RAM usage. 

https://www.reddit.com/r/feedthebeast/comments/5jhuk9/modded_mc_and_memory_usage_a_history_with_a/

**Comment by AtPMonster** [comment]
> I'm not sure if what I said previously wasn't communicated well but let me try again:
> 
> Are you familiar with Garbage Collection with Java? The freezes you are mentioning are most likely the garbage collection process which is a normal process. The more RAM you dedicate the worse it gets. There will be more RAM that has to be dumped and your freezes will continuously get worse. You think you are fixing the problem by dedicating more but, in reality, it's not helping you. It's like having to clear the cache on a phone or tablet, the larger the cache the more time it will take.
> 
> Seeing the 99% of allocated RAM does not necessarily indicate that you need to increase the RAM. If you watch the RAM using the F3 menu, it will climb up then garbage collection happens. It will fall back down and continue to do this over the course of your playthrough. If you dedicate it, Minecraft will use it. This pack can be run with 4GB and be just fine. Again, I would suggest 8GB. The memory is reserved when you dedicate it. Meaning, it will show as being a higher percentage but that's not indicative of you having to give it more.
> 
> To further clarify, those arguments are written by someone who not only works on the modding API (Forge) for minecraft but also works at the company that makes Java. I would urge you to read how the arguments work and learn a bit more about them because I think you are misunderstanding what they are doing. Of course, not every solution will work for everyone. Having said that, these arguments are meant to fix the EXACT problem you are experiencing with RAM usage.
> 
> https://www.reddit.com/r/feedthebeast/comments/5jhuk9/modded_mc_and_memory_usage_a_history_with_a/

I do understand them buddy. Im testing them again and will tell you about the experience - So far these augments set the RAM to 8 GB anyways, so we`ll see how it works :-) Thanks

**Comment by AtPMonster** [comment]
> I'm not sure if what I said previously wasn't communicated well but let me try again:
> 
> Are you familiar with Garbage Collection with Java? The freezes you are mentioning are most likely the garbage collection process which is a normal process. The more RAM you dedicate the worse it gets. There will be more RAM that has to be dumped and your freezes will continuously get worse. You think you are fixing the problem by dedicating more but, in reality, it's not helping you. It's like having to clear the cache on a phone or tablet, the larger the cache the more time it will take.
> 
> Seeing the 99% of allocated RAM does not necessarily indicate that you need to increase the RAM. If you watch the RAM using the F3 menu, it will climb up then garbage collection happens. It will fall back down and continue to do this over the course of your playthrough. If you dedicate it, Minecraft will use it. This pack can be run with 4GB and be just fine. Again, I would suggest 8GB. The memory is reserved when you dedicate it. Meaning, it will show as being a higher percentage but that's not indicative of you having to give it more.
> 
> To further clarify, those arguments are written by someone who not only works on the modding API (Forge) for minecraft but also works at the company that makes Java. I would urge you to read how the arguments work and learn a bit more about them because I think you are misunderstanding what they are doing. Of course, not every solution will work for everyone. Having said that, these arguments are meant to fix the EXACT problem you are experiencing with RAM usage.
> 
> https://www.reddit.com/r/feedthebeast/comments/5jhuk9/modded_mc_and_memory_usage_a_history_with_a/

So I tried it now for almost 2 days. Im getting much more lag spikes and freezes then I did before. Any advice? @Darkosto
